### PR TITLE
Feat/met 5489 angular update

### PR DESCRIPTION
**Pull request by andyjmaclean**

### CODE REVIEW

**Review by jeortizquan — COMMENTED** [review]
I see the process of updating to dependency injection. and a small suggestion for the sandbox.

File: projects/sandbox/src/app/chart/pie/pie.component.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <li
     *ngFor="let item of legendItems; let i = index"
     class="legend-item"
-    title="{{ pieDimension | formatTierDimension }}: {{ item.text }} / {{
+    title="{{ pieDimension | formatTierDimension }} {{ item.text }}:    {{
```
[the remark this answers is not in the document]
Comment (andyjmaclean) [reply projects/sandbox/src/app/chart/pie/pie.component.html:5]: You have a point, @jeortizquan, but this is configured in the ts and could maybe happen in a future ticket?

File: package-lock.json
[the remark this answers is not in the document]
Comment (andyjmaclean) [reply package-lock.json:1]: This is not a dependency that I specify directly in `package.json` but something flagged by the audit, and it's the norm that `npm audit` throws up this kind of thing.

The dependency comes from cypress so I ran ng update cypress, which takes it from version 10.x to version 16.x locally, but this creates issues I was not able to solve: 

image
 
I was able to have the test run using `cypress@11.0`, but not `@^12.0` without encountering this issue. The `@cypress/request` warning is no longer present in the audit (but other things are).

[the remark this answers is not in the document]
Comment (jeortizquan) [reply package-lock.json:1]: I took a look on it, here is a PR https://github.com/europeana/metis-ui/pull/804 to update to cypress `@^13.3.0` you can merge it into this current branch so the dependency is updated. please have a look.

File: projects/sandbox/src/app/chart/pie/pie.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeortizquan) [reply projects/sandbox/src/app/chart/pie/pie.component.html:5]: yes. I agree. I created this followup ticket to address this topic.
https://europeana.atlassian.net/browse/MET-5518

**Review by jeortizquan — CHANGES_REQUESTED** [review]
Please have a look to the suggested PR #804  . In the newest version of cypress the removed `cy.server()`, `cy.route()` https://docs.cypress.io/guides/references/changelog I have updated it using `cy.intercept()` and removed `cy.server()` that now is a default as they mention in the documentation.

**Review by jeortizquan — APPROVED** [review]

File: projects/metis/src/app/dataset/dataset.component.ts [hunk 2]
```diff
@@ -31,15 +31,11 @@ import { WorkflowHeaderComponent } from './workflow/workflow-header';
...
-    private readonly workflows: WorkflowService,
-    private readonly route: ActivatedRoute,
-    private readonly router: Router,
-    private readonly documentTitleService: DocumentTitleService
-  ) {
-    super();
-  }
+  private readonly datasets = inject(DatasetsService);
+  private readonly workflows = inject(WorkflowService);
+  private readonly route = inject(ActivatedRoute);
```
Comment (jeortizquan) [inline projects/metis/src/app/dataset/dataset.component.ts:36]: `route` => `activatedroute`

File: projects/metis/src/app/dataset/datasetform/datasetform.component.ts [hunk 3]
```diff
@@ -33,6 +33,13 @@ const DATASET_TEMP_LSKEY = 'tempDatasetData';
   styleUrls: ['./datasetform.component.scss']
 })
 export class DatasetformComponent extends SubscriptionManager implements OnInit {
+  private readonly authenticationServer = inject(AuthenticationService);
+  private readonly countries = inject(CountriesService);
+  private readonly datasets = inject(DatasetsService);
+  private readonly router = inject(Router);
+  private readonly fb = inject(NonNullableFormBuilder);
```
Comment (jeortizquan) [inline projects/metis/src/app/dataset/datasetform/datasetform.component.ts:40]: `fb` => `formbuilder`

File: projects/metis/src/app/dataset/depublication/depublication.component.ts [hunk 4]
```diff
@@ -29,6 +29,10 @@ import { DepublicationRowComponent } from './depublication-row';
   styleUrls: ['./depublication.component.scss']
 })
 export class DepublicationComponent extends DataPollingComponent {
+  private readonly modalConfirms = inject(ModalConfirmService);
+  private readonly depublications = inject(DepublicationService);
+  private readonly fb = inject(NonNullableFormBuilder);
```
Comment (jeortizquan) [inline projects/metis/src/app/dataset/depublication/depublication.component.ts:34]: `fb`=>`formbuilder`

File: projects/metis/src/app/dataset/workflow/workflow.component.ts [hunk 5]
```diff
@@ -42,13 +43,9 @@ import { WorkflowFormFieldComponent } from './workflow-form-field';
...
 export class WorkflowComponent extends SubscriptionManager implements OnInit {
-  constructor(
-    private readonly workflows: WorkflowService,
-    private readonly fb: FormBuilder,
-    private readonly translate: TranslateService
-  ) {
-    super();
-  }
+  private readonly workflows = inject(WorkflowService);
+  private readonly fb = inject(FormBuilder);
```
Comment (jeortizquan) [inline projects/metis/src/app/dataset/workflow/workflow.component.ts:47]: `fb`=> `formbuilder`

**Review by jeortizquan — COMMENTED** [review]
there are some suggestions for some name variables.

**Review by jeortizquan — COMMENTED** [review]
There are some suggestions for the name of some variables.

File: projects/metis/src/app/dataset/workflow/workflow.component.ts (hunk 5 above)
Comment (andyjmaclean) [reply projects/metis/src/app/dataset/workflow/workflow.component.ts]: Done (but camel-cased!)

File: projects/metis/src/app/dataset/depublication/depublication.component.ts (hunk 4 above)
Comment (andyjmaclean) [reply projects/metis/src/app/dataset/depublication/depublication.component.ts]: Done (but camel-cased!)

File: projects/metis/src/app/dataset/dataset.component.ts (hunk 2 above)
Comment (andyjmaclean) [reply projects/metis/src/app/dataset/dataset.component.ts]: Done (but camel-cased!)

File: projects/sandbox/src/app/upload/upload.component.ts [hunk 6]
```diff
@@ -24,6 +24,12 @@ import { SandboxService } from '../_services';
   styleUrls: ['./upload.component.scss']
 })
 export class UploadComponent extends DataPollingComponent {
+  private readonly fb = inject(FormBuilder);
```
[the remark this answers is not in the document]
Comment (andyjmaclean) [reply projects/sandbox/src/app/upload/upload.component.ts]: Done (but camel-cased!)

File: projects/sandbox/src/app/sandbox-navigation/sandbox-navigation.component.ts [hunk 7]
```diff
@@ -46,6 +46,11 @@ enum ButtonAction {
   styleUrls: ['/sandbox-navigation.component.scss']
 })
 export class SandboxNavigatonComponent extends DataPollingComponent implements OnInit {
+  private readonly fb = inject(NonNullableFormBuilder);
```
[the remark this answers is not in the document]
Comment (andyjmaclean) [reply projects/sandbox/src/app/sandbox-navigation/sandbox-navigation.component.ts]: Done (but camel-cased!)

**Pull request merged**